Library/Homebrew/tab.rb

```diff
@@ -242,6 +242,26 @@ def compiler
     super || DevelopmentTools.default_compiler
   end
 
+  def homebrew_tag
+    homebrew_version.sub(/\-\d+\-g([a-f0-9]+)(?:\-dirty)?\Z/, "")
+  end
+
+  def parsed_homebrew_version
+    return Version::NULL if homebrew_version.nil?
+    Version.new(homebrew_tag)
+  end
+
+  # Whether there is reliable runtime dependency information in the receipt.
+  def reliable_runtime_dependencies?
```

Comment:
I think clean core code trumps clean test code. We should make test code as good as possible but it's there to serve the core code (and us!) and not vice-versa.